ons-spark/ancillary-topics/pandas_udfs_rescue.ipynb

```diff
@@ -0,0 +1,902 @@
+    "\n",
+    "### Declaring Pandas UDFs in Spark\n",
+    "\n",
+    "You can create a Pandas UDF in a number of ways.\n",
+    "- Using the `@pandas_udf(<type>, F.PandasUDFType.<type>)` decorator.\n",
+    "- Assigning the UDF using `<udf_name>_udf = F.pandas_udf(<udf_name>, returnType = <type>)`\n",
+    "- Using Python Type hints. Python hints can now be added to enable Pandas UDFs to be more descriptive. For more info see [here](https://www.databricks.com/blog/2020/05/20/new-pandas-udfs-and-python-type-hints-in-the-upcoming-release-of-apache-spark-3-0.html).\n",
+    "\n",
+    "This demo will use thr `pandas_udf` decorator and Python hints to annotate the functions with the input and output data types.\n",
+    "\n",
+    "\n",
+    "### Spark Setup for Pandas UDFs\n",
+    "\n",
+    "To use Pandas UDFs you will have to install PyArrow using:\n",
+    "```\n",
+    "pip install PyArrow\n",
+    "```\n",
+    "\n",
+    "You should also add configs to enable PyArrow optimisation and fallback if it is not installed. Even if you are not using Pandas UDFs the below configs can make the conversion between Pandas and Spark more efficient.\n",
+    "\n",
+    "```\n",
+    ".config('spark.sql.execution.arrow.enabled', 'true')\n",
+    ".config('spark.sql.execution.arrow.fallback.enabled', 'true')\n",
+    "```\n",
+    "\n",
+    "For Spark 2.4 you will need to add the below additional configs to your Spark session for compatibility:\n",
+    "\n",
+    "```\n",
+    ".config('spark.excutorEnv.ARROW_PRE_0_15_IPC_FORMAT', 1)\n",
+    ".config('spark.workerEnv.ARROW_PRE_0_15_IPC_FORMAT', 1)\n",
+    "```\n",
+    "\n",
+    "In our demo we are using a local Spark session and will be working with the animal rescue dataset used throughout in this book."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import packages and set up Spark session for PyArrow optimisation and compatibility with Spark 2.4\n",
+    "\n",
+    "from pyspark.sql import SparkSession\n",
+    "import pyspark.sql.functions as F\n",
+    "from pyspark.sql.functions import pandas_udf\n",
+    "import pyspark.sql.types as T\n",
+    "\n",
+    "import pandas as pd\n",
+    "\n",
+    "spark = (SparkSession.builder.master(\"local[2]\")\n",
+    "         .appName(\"pandas_udfs\")\n",
+    "         .config('spark.sql.execution.arrow.enabled', 'true')\n",
+    "         .config('spark.sql.execution.arrow.fallback.enabled', 'true')\n",
+    "         .config('spark.excutorEnv.ARROW_PRE_0_15_IPC_FORMAT', 1)\n",
+    "         .config('spark.workerEnv.ARROW_PRE_0_15_IPC_FORMAT', 1)\n",
+    "         .getOrCreate())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read in our rescue dataset\n",
+    "\n",
+    "rescue_path_csv = '/training/animal_rescue.csv'\n",
```

Comment:
use the config file for file paths, see other notebooks in the book

Reply:
I see that you've changed the path to rescue_path_csv in the config previously, maybe change that back so they're consistent?